pallets/phala/src/compute/wrapped_balances.rs

```diff
@@ -133,6 +144,217 @@ pub mod pallet {
+		}
+
+		fn withdraw(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+			_reasons: WithdrawReasons,
+			_liveness: ExistenceRequirement,
+		) -> Result<Self::NegativeImbalance, DispatchError> {
+			Ok(Self::NegativeImbalance::zero())
+		}
+
+		fn make_free_balance_be(
+			_who: &T::AccountId,
+			_value: Self::Balance,
+		) -> SignedImbalance<Self::Balance, Self::PositiveImbalance> {
+			SignedImbalance::Positive(Self::PositiveImbalance::zero())
+		}
+	}
+
+	impl<T: Config> ReservableCurrency<T::AccountId> for Pallet<T>
+	where
+		BalanceOf<T>: sp_runtime::traits::AtLeast32BitUnsigned + Copy + FixedPointConvert + Display,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + pallet_balances::Config + vault::Config,
+		T: pallet_balances::Config<Balance = BalanceOf<T>>,
+	{
+		fn can_reserve(who: &T::AccountId, value: Self::Balance) -> bool {
+			Self::total_balance(who) >= value
+		}
+		fn reserved_balance(who: &T::AccountId) -> Self::Balance {
+			ElectionReserves::<T>::get(who).unwrap_or_default()
+		}
+		fn reserve(who: &T::AccountId, value: Self::Balance) -> DispatchResult {
+			let actual = value.min(Self::total_balance(who));
+			ElectionReserves::<T>::mutate(who, |reserve| {
+				*reserve = Some(reserve.unwrap_or_default() + actual);
+			});
+			Ok(())
+		}
+		fn unreserve(who: &T::AccountId, value: Self::Balance) -> Self::Balance {
+			let actual = value.min(ElectionReserves::<T>::get(who).unwrap_or_default());
+			ElectionReserves::<T>::mutate(who, |reserve| {
+				*reserve = Some(reserve.unwrap_or_default() - actual);
+			});
+			value - actual
+		}
+		fn slash_reserved(
+			who: &T::AccountId,
+			value: Self::Balance,
+		) -> (Self::NegativeImbalance, Self::Balance) {
+			let actual = value.min(ElectionReserves::<T>::get(who).unwrap_or_default());
+			ElectionReserves::<T>::mutate(who, |reserve| {
+				*reserve = Some(reserve.unwrap_or_default() - actual);
+			});
+			Self::burn_from(who, actual)
+				.expect("burn nft when slash reserved should success; qed.");
+			let imbalance = <T as PhalaConfig>::Currency::withdraw(
+				&T::WrappedBalancesAccountId::get(),
+				actual,
+				WithdrawReasons::all(),
+				ExistenceRequirement::AllowDeath,
+			)
+			.expect("slash imbalance should success; qed.");
+			(imbalance, value - actual)
+		}
+		fn repatriate_reserved(
+			_slashed: &T::AccountId,
+			_beneficiary: &T::AccountId,
+			_value: Self::Balance,
+			_status: BalanceStatus,
+		) -> Result<Self::Balance, DispatchError> {
+			Ok(Zero::zero())
+		}
+	}
+	impl<T: Config> LockableCurrency<T::AccountId> for Pallet<T>
+	where
+		BalanceOf<T>: sp_runtime::traits::AtLeast32BitUnsigned + Copy + FixedPointConvert + Display,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + pallet_balances::Config + vault::Config,
+		T: pallet_balances::Config<Balance = BalanceOf<T>>,
+	{
+		type Moment = T::BlockNumber;
+		type MaxLocks = T::MaxLocks;
+		fn set_lock(
+			_id: LockIdentifier,
```

Comment:
Would be nice to double check the id for robustness

Reply:
done